src/widgetastic_patternfly/__init__.py

```diff
@@ -1849,3 +1850,179 @@ def click_location(self, name, handle_alert=True):
+            for el in self.browser.elements(self.DATES):
+                if not bool({'old', 'new', 'disabled'} & self.browser.classes(el)):
+                    dates.update({int(el.text): el})
+            return dates
+
+    @View.nested
+    class month_pick(HeaderView):   # noqa
+        MONTHS = ".//*[contains(@class, 'datepicker-months')]/table/tbody/tr/td/*"
+
+        @property
+        def _elements(self):
+            months = {}
+            for el in self.browser.elements(self.MONTHS):
+                if not bool({'disabled'} & self.browser.classes(el)):
+                    months.update({el.text: el})
+            return months
+
+    @View.nested
+    class year_pick(HeaderView):    # noqa
+        YEARS = ".//*[contains(@class, 'datepicker-years')]/table/tbody/tr/td/*"
+
+        @property
+        def _elements(self):
+            years = {}
+            for el in self.browser.elements(self.YEARS):
+                if not bool({'old', 'new', 'disabled'} & self.browser.classes(el)):
+                    years.update({int(el.text): el})
+            return years
+
+        def select(self, value):
+            for _yr in range(10):
+                start_yr, end_yr = [int(item) for item in self.datepicker_switch.read().split('-')]
+                if start_yr <= value <= end_yr:
+                    for el, web_el in self._elements.items():
+                        if el == value:
+                            web_el.click()
+                            return True
+                elif value < start_yr:
+                    self.prev_button.click()
+                elif value > end_yr:
+                    self.next_button.click()
+            else:
+                raise ValueError("Not valid year or with more than century difference")
+
+    def read(self):
+        """Read the current date form DatePicker
+
+        Returns:
+            :py:class:`datetime`
+        """
+        try:
+            return datetime.strptime(self.textbox.value, self.strptime_format)
+        except ValueError:
+            return None
+
+    def fill(self, value):
+        """Fill date to DatePicker
+
+        Args:
+           value: datetime object.
+
+        Returns:
+            :py:class:`bool`
+        """
+        current_date = self.read()
+        if current_date:
+            if value.date() == current_date.date(): return False
+
+        if not self.is_readonly:
+            date = datetime.strftime(value, self.strptime_format)
+            self.textbox.fill(date)
+            self.date_pick._elements[self.date_pick.active].click()
+            return True
+        else:
+            self.browser.click(self.textbox)
+            sw_value = self.date_pick.datepicker_switch.read().split(' ')
+            if (value.strftime("%B") not in sw_value) or (str(value.year) not in sw_value):
+                self.date_pick.datepicker_switch.click()
+                if str(value.year) is not self.month_pick.datepicker_switch.read():
```

Comment:
What do you check here ? "is" is used to check if both vars refer to the same object.

BTW, I doubt that date picker works too slow if you just change value each time.
So, I believe you can remove most these checks above. You need to check whether value and read are different and that's all. If those are different go and pick year, month, date.

Reply:
ok, I like to remove all checks.